File: 191.Number of 1 Bits.cc
// https://leetcode.com/problems/number-of-1-bits/

class Solution {
public:
    int hammingWeight(uint32_t n) {
        int amount = 0;
        while (n != 0)
        {
            amount += n & 0x1;
            n >>= 1;
        }
        return amount;
    }
};

File: leetcode/0-300/151-200/191.Number of 1 Bits.rs
// https://leetcode.com/problems/number-of-1-bits

impl Solution {
    pub fn hammingWeight (n: u32) -> i32 {
        let mut num = n;
        let mut weight = 0;
        while num > 0 {
            if num & 0x1 == 0x1 {
                weight += 1;
            }
            num >>= 1;
        }
        weight
    }
}